sdk/identity/Azure.Identity/src/DefaultAzureCredentialFactory.cs

```diff
@@ -37,65 +38,68 @@ public TokenCredential[] CreateCredentialChain()
                 return s_defaultCredentialChain;
             }
 
-            int i = 0;
-            TokenCredential[] chain = new TokenCredential[9];
+            List<TokenCredential> chain = new(9);
 
             if (!Options.ExcludeEnvironmentCredential)
             {
-                chain[i++] = CreateEnvironmentCredential();
+                chain.Add(CreateEnvironmentCredential());
             }
 
             if (!Options.ExcludeManagedIdentityCredential)
             {
-                chain[i++] = CreateManagedIdentityCredential();
+                chain.Add(CreateManagedIdentityCredential());
             }
 
             if (!Options.ExcludeAzureDeveloperCliCredential)
             {
-                chain[i++] = CreateAzureDeveloperCliCredential();
+                chain.Add(CreateAzureDeveloperCliCredential());
             }
 
             if (!Options.ExcludeSharedTokenCacheCredential)
             {
-                chain[i++] = CreateSharedTokenCacheCredential();
+                chain.Add(CreateSharedTokenCacheCredential());
             }
 
             if (!Options.ExcludeVisualStudioCredential)
             {
-                chain[i++] = CreateVisualStudioCredential();
+                chain.Add(CreateVisualStudioCredential());
             }
 
             if (!Options.ExcludeVisualStudioCodeCredential)
             {
-                chain[i++] = CreateVisualStudioCodeCredential();
+                chain.Add(CreateVisualStudioCodeCredential());
             }
 
             if (!Options.ExcludeAzureCliCredential)
             {
-                chain[i++] = CreateAzureCliCredential();
+                chain.Add(CreateAzureCliCredential());
             }
 
             if (!Options.ExcludeAzurePowerShellCredential)
             {
-                chain[i++] = CreateAzurePowerShellCredential();
+                chain.Add(CreateAzurePowerShellCredential());
             }
 
             if (!Options.ExcludeInteractiveBrowserCredential)
             {
-                chain[i++] = CreateInteractiveBrowserCredential();
+                chain.Add(CreateInteractiveBrowserCredential());
             }
 
-            if (i == 0)
+            if (chain.Count == 0)
             {
                 throw new ArgumentException("At least one credential type must be included in the authentication flow.", "options");
             }
 
-            return chain;
+            return chain.ToArray();
         }
 
         public virtual TokenCredential CreateEnvironmentCredential()
         {
-            return new EnvironmentCredential(Pipeline, Options);
+            return new EnvironmentCredential(Pipeline, new EnvironmentCredentialOptions
+            {
+                AuthorityHost = Options.AuthorityHost,
+                DisableInstanceDiscovery = Options.DisableInstanceDiscovery,
+            });
```

Comment:
If I recall there was some special handling here when it was passing in `Options` directly to ensure `AdditionallyAllowedTenants` was honored properly, and the items in the list weren't duplicated from what was in the environment variable. Are we sure this has this same functionality?

Reply:
I think we have it covered now